data/index.js

```diff
@@ -24,13 +24,70 @@ const selectors = {};
 const actions = {};
 let listeners = [];
 
+/**
+ * Higher-order function creator to create a function which invokes ifTrueFn
+ * only if the given predicate returns a truthy value.
+ *
+ * @param {Function} predicate Predicate test.
+ *
+ * @see http://ramdajs.com/docs/#when
+ *
+ * @return {Function} Higher-order function to call with ifTrueFn.
+ */
+const when = ( predicate ) => ( ifTrueFn ) => ( ...args ) => {
+	if ( predicate() ) {
+		ifTrueFn( ...args );
+	}
+};
+
+/**
+ * Higher-order function creator to create a function which invokes ifTrueFn
+ * only if the given predicate returns a truthy value, with the result.
+ *
+ * @param {Function} predicate Predicate test returning tuple of isTrue, result
+ *
+ * @return {Function} Higher-order function to call with ifTrueFn.
+ */
+const whenWith = ( predicate ) => ( ifTrueFn ) => () => {
+	const [ isTrue, result ] = predicate();
+	if ( isTrue ) {
+		ifTrueFn( result );
+	}
+};
+
 /**
  * Global listener called for each store's update.
  */
 export function globalListener() {
 	listeners.forEach( listener => listener() );
 }
 
+/**
+ * Convenience for registering reducer with actions and selectors.
+ *
+ * @param {string} reducerKey Reducer key.
+ * @param {Object} options    Store description (reducer, actions, selectors).
+ *
+ * @return {Object} Registered store object.
+ */
+export function registerStore( reducerKey, options ) {
```

Comment:
I like it 👍

Reply:
Yes, much nicer. I think this should be the only public API available. We could turn `registerReducer`, `registerActions` and `registerSelectors` into the implementation detail.

Reply:
Unless we want to allow other plugins to make modifications to not their own stores.